problems/missing-number/missing-number.rs

```diff
@@ -0,0 +1,43 @@
+//returns the missing number, or in the case Vec is empty, returns None
+fn missing_number(numbers: &[isize]) -> Option<isize> {
+  let mut numbers = Vec::from(numbers);
+  numbers.sort();
+  let sum: isize = numbers.iter().sum();
+  let incremental_sum = ((numbers.len() + 1) as isize * (numbers.last()? + numbers.get(0)?)) / 2;
+  Some(incremental_sum - sum)
+}
+
+
+fn main() {
+  println!("missing number: {}", missing_number(&[1,3,5,7,11]).unwrap());
```

Comment:
This gives 9 but 2, 4, 6, 8, 9, and 10 are missing. The input is actually incorrect. I believe that all solutions suffer from a misinterpretation of the problem.